src/charm.py

```diff
@@ -23,6 +23,7 @@
 LOGGER = logging.getLogger(__name__)
 _INGRESS_SUB_REGEX = re.compile("[^0-9a-zA-Z]")
 BOOLEAN_CONFIG_FIELDS = ["rewrite-enabled"]
+CREATED_BY_LABEL = "app.juju.is/created-by="
```

Comment:
As discussed in the daily digest, I don't understand where this label comes from.
The charm uses python kubernetes-client library and doesn't explicitly set this label which means:
- Either the library takes it from the pod calling the API (I don't think that's the case, I quickly took a look at the library code and didn't find such business logic)
- Or it's added by Juju for every ressource within the namespace (perhaps edited automatically by the modelcontroller ? But how does juju show who created this kubernetes ressources as there could be multiple charms deployed in a model)

This should be clarified for 2 reasons:
- Ensure we won't delete services/ingress we don't want to delete
- Ensure this label won't be removed in a future releases

One approach could be to add our own label in this charm when creating the Ingress/Service, the drawback is that we won't be able to cleanup previously created ressources (in case of an upgrade of charm)

Reply:
After some digging, I think I found out how Juju knows which application created the resource.

When Juju receives a request via cassadmission, fetches the juju application name associated with a service account that in its turn has this value set in the label "app.kubernetes.io/name".
 
Code:
https://github.com/juju/juju/blob/develop/worker/caasadmission/handler.go#L131

microk8s output:
```
$ microk8s kubectl get serviceaccount -n indico nginx1 --show-labels
NAME     SECRETS   AGE     LABELS
nginx1   0         4h48m   app.kubernetes.io/managed-by=juju,app.kubernetes.io/name=nginx1
$ microk8s kubectl get ing -n indico --show-labels
NAME                    CLASS    HOSTS           ADDRESS     PORTS   AGE     LABELS
indico-local-ingress    public   indico.local    127.0.0.1   80      4h46m   app.juju.is/created-by=nginx1
hello-kubecon-ingress   public   hello-kubecon   127.0.0.1   80      6m5s    app.juju.is/created-by=nginx2
```

Reply:
Ah makes sens, it uses an Admission Controller to get the events, and uses the Service Account attached to the pod.
Thx for digging, good to know anyway :)

Did you get some feedback from Juju team on this topic ?

Reply:
Yes, here it is:
https://chat.charmhub.io/charmhub/pl/3ik6j6ryrjby7ykjx8fmy9keha

```
I can't guarantee these won't change in the future. We'll try to ensure they don't get changed, but I would recommend implementing a test on your end.

Why I could see this changing at some point is two things, labelling of resources created by the app isn't perfect, I think there is still some edge cases to work through; secondly the app name isn't unique, so we need to add a model-uuid into it, since resources could be created outside the model managed namespace.
```

Reply:
We could work with both options. First, we could look for our label. If doesn't exist then we look for the app.juju.is/created-by=. If none of them exist, then nothing will be deleted. What do you think?

Reply:
Well not sure the added business logic worth it
- Having our own label, will never trigger the fallback to "created-by", unless someone removes it from our code. In that case we should ensure a unit/integration test will fail due to broken feature
- Relying on juju "created-by" label, will only break if they remove it, and here once more we should have a unit/integration tests that will fail if that's the case.

I'm more in favor of the 1st one because we don't run our integration tests scheduled, which means that the feature could be broken without us noticing until we do a code change that breaks the integration tests.

Reply:
We could work on this in a new story if it's ok for you. And for this one, we continue using the Juju label.